Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/Rooms/IRoomProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.Multiplayer.Connections.Rooms;
+using DCL.Multiplayer.Connections.Rooms.Connective;
+using LiveKit.Rooms;
+using System.Threading;
+
+namespace DCL.Multiplayer.Connections.Archipelago.Rooms
+{
+    /// <summary>
+    /// This interface is not connected to <see cref="IConnectiveRoom"/>.
+    /// It has some common fields but its contract can mutate independently
+    /// </summary>
+    public interface IRoomProvider
```

Comment:
What's the point of IRoomProvider? It doesn't jst "Provide" or it's not being a just source of IRoom, it acts as an object that handles the lifecycle of the room

Reply:
It has one single purpose: `Constructs either a fixed connective room or a dynamic one based on the archipelago.`
Regarding the name can you propose a better one? It's not necessary to go by book, anything that would feat its purpose.

Reply:
The thing is it's not the `Room` itself, it's an upper layer that provides one of the rooms according to the inner conditions